receive_buffer.go

```diff
@@ -106,24 +102,29 @@ func (buf *receiveBuffer) read(b []byte, deadline time.Time) (totalN int, err er
 				return
 			}
 			timer := time.NewTimer(deadline.Sub(now))
+			stopTimer := func() {
+				if !timer.Stop() {
```

Comment:
Drive-by, noticed that I wasn't closing the timer correctly.

n.b. I hate this API, `timer.Stop()` should just do the right thing.

Reply:
Agreed, `timer.Stop` should drain the channel.  I can't think of a good reason that it wouldn't.

However, I don't think draining the channel is necessary here.  `timer` is getting thrown away anyway, right?  So it doesn't matter if there's still a value on its channel.

Reply:
Ahh, you made me realize that I don't even need to call stop(), because there's no goroutine to stop. I'll fix.

Reply:
Fixed.